### ISSUE TITLE
High memory usage after having a extened session

### ISSUE THREAD

**Issue by ChihayaK** [tlkh/asitop#6]
I relized this problem when I forgot to terminate the asitop and let it run overnight. The next day I relized the some program is using a huge amount of memory. After terminated the asitop it released nearly 15 gigs of memory. Is this the inteded behavior because of the nature of logging?

**Comment by tlkh** [comment]
Yeah it could be the case that the `powermetrics` process started by `asitop` could have been accumulating memory. I don't think asitop itself will use so much memory, it only logs maximum 30s of power information. 

Although it is not intended, if the issue is `powermetrics`, I don't have a solution for that short of restarting the process at an interval of maybe a few hours.

**Comment by ChihayaK** [comment]
After read though the man page of [powermatrics](https://www.unix.com/man-page/osx/1/powermetrics/) I guess this could be eliminated by giving an -n argument to limit its periodic sample instead of the default unlimited samples. But i am just gessing, I will do some testing tonight.

**Comment by tlkh** [comment]
Yes, previously I had just called powermetrics with -n1, it was the most logical to me at that time. However, every time the powermetrics process starts and stops, there is a overhead that results in a performance hit (for example, if you are doing any form of benchmarking.

So probably there should be a kind of balance...

**Comment by Vlor999** [comment]
Hi everyone, since this project seems to be unmaintained and has accumulating bugs with recent macOS updates, I've created an actively maintained fork called mxtop. It includes pending community fixes and is fully working on the latest Apple Silicon chips.
You can install it right now using: pip install mxtop The new repository is here: https://github.com/Vlor999/mxtop
Feel free to open issues or PRs there!
